pkg/query/endpointset.go

```diff
@@ -0,0 +1,704 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package query
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/thanos-io/thanos/pkg/exemplars/exemplarspb"
+	"github.com/thanos-io/thanos/pkg/info/infopb"
+	"google.golang.org/grpc"
+
+	"github.com/thanos-io/thanos/pkg/component"
+	"github.com/thanos-io/thanos/pkg/metadata/metadatapb"
+	"github.com/thanos-io/thanos/pkg/rules/rulespb"
+	"github.com/thanos-io/thanos/pkg/runutil"
+	"github.com/thanos-io/thanos/pkg/store"
+	"github.com/thanos-io/thanos/pkg/store/labelpb"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+	"github.com/thanos-io/thanos/pkg/targets/targetspb"
+)
+
+const (
+	unhealthyEndpointMessage = "removing endpoint because it's unhealthy or does not exist"
+
+	// Default minimum and maximum time values used by Prometheus when they are not passed as query parameter.
+	MinTime = -9223309901257974
+	MaxTime = 9223309901257974
+)
+
+type EndpointSpec interface {
```

Comment:
Is there any other implementation of this interface than grpcEndpointSpec?

Reply:
No. only `grpcEndpointSpec` is there.

Reply:
We are not creating a mock implementation for the tests. I agree with you about removing it.